build_runner/lib/src/generate/build_impl.dart

```diff
@@ -433,26 +433,26 @@ class BuildImpl {
     Digest inputsDigest;
 
     for (var output in declaredOutputs) {
+      var wasOutput = writer.assetsWritten.contains(output);
+      var digest = wasOutput ? await _reader.digest(output) : null;
       var node = _assetGraph.get(output) as GeneratedAssetNode;
-      node
-        ..needsUpdate = false
-        ..wasOutput = false
-        ..lastKnownDigest = null
-        ..globs = reader.globsRan.toSet();
 
+      var allInputs = reader.assetsRead;
+      if (node.primaryInput != null) allInputs.add(node.primaryInput);
+      inputsDigest ??= await _computeCombinedDigest(node.inputs, reader);
+
+      // **IMPORTANT**: All updates to `node` must be synchronous. With lazy
```

Comment:
...meaning one should change all of the properties in one go?

Reply:
yes